app/src/main/jni/cve20160808.c

```diff
@@ -0,0 +1,49 @@
+#include <jni.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <android/log.h>
+
+int checkIsVulnerable(){
+	FILE *file;
+	long size;
+	uint8_t *buffer;
+	int result;
+	
+
+	file = fopen("/system/lib/libminikin.so", "rb");
+
+	if(file == NULL){
+		result = 0;
+		goto done;	
+	}
+
+	fseek(file, 0, SEEK_END);
+	size = ftell(file);
+	rewind(file);
+
+	buffer = (uint8_t *)malloc(sizeof(char)*size);
+
+	fread(buffer, 1, size, file);
+	uint8_t needle[4] = {0x53, 0x55, 0x55, 0x15};
```

Comment:
What does this needle represent? some information about this, what the bug was, and how the patch works would be great.